### PR TITLE
feat: Refactor handling of project version numbers and tags

### PR DESCRIPTION
**Pull request by bdferris-v2**
Per issue #1139, refactor how versions are handled in project.

Currently, project version info is passed in via a `versionTag` env variable, which is only specified in the Github workflow environment.  Consequently, `project.version` ends up being `null` when run locally.  This causes weirdness for tools that expect a valid version number.

This change switches the project to use the axion-release plugin, which determines the project version by looking at the git repo's tags.

Along with that switch, I've moved the project group and version specification into the root build.gradle allprojects {} such that these values do not need to be specified in each sub-poject.

I also did some related clean-up for our jitpack configuration (and Maven repository modules, should we every decide to publish them).  I believe our current jitpack config doesn't actually work correctly because we only publish artifacts from the main package, but not core.

You can see this in practice when looking at:
https://jitpack.io/com/github/MobilityData/gtfs-validator/3.0.1/gtfs-validator-3.0.1.pom

Notice how it has a dependency on the `core` module but we don't actually publish it anywhere.  If you actually attempt to use our project as a jitpack dependency, it will fail.

The change was to also publish the core module and to specify more natural artifact names.

### CODE REVIEW

**Comment by bdferris-v2** [comment]
(This isn't ready for review yet.  I just wanted to see what would happen with all the workflow tasks.)

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -45,8 +45,6 @@ jobs:
           restore-keys: ${{ runner.os }}-gradle
       - name: Run Java tests
         uses: eskatos/gradle-command-action@v1
-        env:
-          versionTag: ${{ steps.prep.outputs.versionTag }}
```
Comment (maximearmstrong) [inline .github/workflows/docker.yml:49]: I agree with removing this. I'm not sure if we need the `Prepare version name` job at all. I see that it used below in two jobs name, but I wonder if we could just get rid of that as well. Thoughts? cc @barbeau

File: .github/workflows/test_pack_doc.yml [hunk 2]
```diff
@@ -102,15 +102,13 @@ jobs:
           restore-keys: ${{ runner.os }}-gradle
       - name: Package cli app jar with Gradle
         uses: eskatos/gradle-command-action@v1
-        env:
-          versionTag: ${{ needs.prepare-version-name.outputs.versionTag }}
```
Comment (maximearmstrong) [inline .github/workflows/test_pack_doc.yml:106]: Same as mentioned in `docker.yml`. We are still using the versionTag a lot, but mostly for job names. I'm not sure if this is relevant anymore if the validator is responsible for its own versioning now? cc @barbeau

**Review by maximearmstrong — COMMENTED** [review]
Thanks @bdferris-v2, this looks good! Some thoughts in-line.

File: .github/workflows/test_pack_doc.yml (hunk 2 above)
Comment (barbeau) [reply .github/workflows/test_pack_doc.yml:106]: Yes, I think we should remove all version tag references from the test_pack_doc GitHub Action? I believe it should be printed to the log as part of the commands shown in the JAR file name?

File: .github/workflows/docker.yml (hunk 1 above)
Comment (barbeau) [reply .github/workflows/docker.yml:49]: I think so - from looking at the script it seems like the commands that set up the Docker image version tag are separate from the script setting the validator version tag.

Comment (bdferris-v2) [reply .github/workflows/docker.yml:49]: I went ahead and dropped all the versioning code.

File: .github/workflows/test_pack_doc.yml (hunk 2 above)
Comment (bdferris-v2) [reply .github/workflows/test_pack_doc.yml:106]: I removed the versionTag references here as well.

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -73,34 +56,7 @@ jobs:
...
-            VERSION=$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
-          elif [[ $GITHUB_REF == refs/pull/* ]]; then
-            VERSION=pr-${{ github.event.number }}
-          fi
-          TAGS="${DOCKER_IMAGE}:${VERSION}"
-          if [ "${{ github.event_name }}" = "push" ]; then
-            TAGS="${DOCKER_IMAGE}:latest"
-          fi
-          echo ::set-output name=version::${VERSION}
-          echo ::set-output name=tags::${TAGS}
```
Comment (barbeau) [inline .github/workflows/docker.yml:90]: I think we need this line that sets up the output for `tags`, as that seems to be used later to set the Docker image tag in line 89:

```
tags: ${{ steps.prep.outputs.tags }}
```

Comment (bdferris-v2) [reply .github/workflows/docker.yml:90]: Ah, I totally missed that.  I restored the TAGS code.

**Review by barbeau — APPROVED** [review]
LGTM

**Comment by barbeau** [comment]
@bdferris-v2 The branch rules require that the main branch be merged into the PR, if you could do that.

**Comment by bdferris-v2** [comment]
Hopefully should be merged at this point.

**Comment by isabelle-dr** [comment]
I forgot to add my review before merging: I approve this PR, thanks!

**Pull request merged**